trainer-client/client.go

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018. IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package client
+
+import (
+	"fmt"
+
+	"github.com/AISphere/ffdl-commons/config"
+	"github.com/AISphere/ffdl-commons/util"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+	"github.com/AISphere/ffdl-trainer/trainer/grpc_trainer_v2"
+)
+
+const (
+	disabled = "disabled"
+	// TrainerV2LocalAddress exposes the local address that is used if we run with DNS disabled
+	TrainerV2LocalAddress = ":30005"
+)
+
+// TrainerClient is a client interface for interacting with the trainer service.
+type TrainerClient interface {
+	Client() grpc_trainer_v2.TrainerClient
+	Close() error
+}
+
+type trainerClient struct {
+	client grpc_trainer_v2.TrainerClient
+	conn   *grpc.ClientConn
+}
+
+// NewTrainer create a new load-balanced client to talk to the Trainer
+// service. If the dns_server config option is set to 'disabled', it will
+// default to the pre-defined LocalPort of the service.
+func NewTrainer() (TrainerClient, error) {
+	return NewTrainerWithAddress(TrainerV2LocalAddress)
+}
+
+// NewTrainerWithAddress create a new load-balanced client to talk to the Trainer
+// service. If the dns_server config option is set to 'disabled', it will
+// default to the pre-defined LocalPort of the service.
+func NewTrainerWithAddress(addr string) (TrainerClient, error) {
+	address := fmt.Sprintf("ffdl-trainer.%s.svc.cluster.local:80", config.GetPodNamespace())
```

Comment:
I think we shouldn't hard code the KUBEDNS service name over here, since DLaaS and FfDL will name their services differently.

Reply:
> since DLaaS and FfDL will name their services differently.

Is that true?  In any case, so I guess this should be a viper/env variable.  Let's keep this in mind, and do as separate PR.  Also, I think we need to figure viper so that these config variables are not pulled from commons.  Let me think about that a bit a make a proposal.

I'm going to go ahead and merge this PR.  Good enough for now.